### DISCUSSION TITLE
How does packasge resulution work in Deno?

### DISCUSSION THREAD

**Discussion by tlaziuk** [denoland/deno#26365] [General]
Hi, I wonder how does package resolution work in Deno

let's use a real example of a package that's really cross-platform: `hono`

so let's say there are imports `npm:hono@^4.6.0` and `jsr:@hono/hono@^4.6.0` - is Deno smart enough to tell that these are the same packages?

**Comment by marvinhagemeister** [comment]
> so let's say there are imports npm:hono@^4.6.0 and jsr:@hono/hono@^4.6.0 - is Deno smart enough to tell that these are the same packages?

No, these are treated as separate packages in Deno at the moment. The topic of deduplicating packages across registries came up a few times in internal discussions, but nothing more than that.
